src/Listener/NoCacheImageRequestSubscriber.php

```diff
@@ -37,7 +37,7 @@ public function onRequest(RequestEvent $requestEvent): void
 
             if ($nativeImage->isNextGen()) {
                 $response = new Response(
-                    (string) $image->encode($nativeImage->getNextGenExtension(), $requestEvent->getQuality()),
+                    (string) $image->encodeByExtension($nativeImage->getNextGenExtension(), $requestEvent->getQuality()),
```

Comment:
We use `$image->encodeByExtension` and `$image->encode` at several places.

We should factorize this to use our own `ImageEncoder` which will support `quality` and `progressive` parameters